leakcanary-android-core/src/main/java/leakcanary/internal/InternalLeakCanary.kt

```diff
@@ -76,6 +83,33 @@ internal object InternalLeakCanary : (Application) -> Unit, OnObjectRetainedList
         onHeapAnalyzedListener = OnHeapAnalyzedListener {}
     )
 
+  internal enum class FormFactor {
+    MOBILE,
+    TV,
+    WATCH,
+  }
+
+  val formFactor by lazy {
+    val currentModeType =
+      (application.getSystemService(UI_MODE_SERVICE) as UiModeManager).currentModeType
```

Comment:
wow, UiModeManager has been here since API lvl 8, first time I see it.

Reply:
I felt the same way :)